lib/whitehall_importer/create_edition.rb

```diff
@@ -184,18 +186,67 @@ def create_revision_history(edition)
         entry_type = history.imported_entry_type(event, edition_number)
         next if entry_type.nil?
 
-        details = TimelineEntry::WhitehallImportedEntry.create!(
-          entry_type: entry_type,
-        )
-        TimelineEntry.create!(
-          entry_type: :whitehall_migration,
-          created_at: event["created_at"],
-          created_by_id: user_ids[event["whodunnit"]],
-          edition: edition,
-          document: edition.document,
-          details: details,
-        )
+        details = create_whitehall_imported_entry(entry_type)
+        create_timeline_entry(details,
+                              edition,
+                              event["created_at"],
+                              user_ids[event["whodunnit"]])
       end
     end
+
+    def create_notes(edition)
+      whitehall_edition["editorial_remarks"].each do |event|
+        contents = {
+          body: event["body"],
+        }
+        details = create_whitehall_imported_entry("internal_note", contents)
+        create_timeline_entry(details,
+                              edition, event["created_at"], event["author_id"])
+      end
+    end
+
+    def create_fact_checks(edition)
+      whitehall_edition["fact_check_requests"].each do |event|
+        create_fact_check_request(edition, event)
+        create_fact_check_response(edition, event) if event["comments"].present?
+      end
+    end
+
+    def create_fact_check_request(edition, event)
+      contents = {
+        email_address: event["email_address"],
+        instructions: event["instructions"],
+      }
+      details = create_whitehall_imported_entry("fact_check_request", contents)
+      create_timeline_entry(details,
+                            edition, event["created_at"], event["requestor_id"])
+    end
+
+    def create_fact_check_response(edition, event)
+      contents = {
+        email_address: event["email_address"],
+        comments: event["comments"],
+      }
+      details = create_whitehall_imported_entry("fact_check_response", contents)
+      create_timeline_entry(details, edition, event["updated_at"])
+    end
+
+    def create_whitehall_imported_entry(entry_type, contents = {})
+      TimelineEntry::WhitehallImportedEntry.create!(
+        entry_type: entry_type,
+        contents: contents,
+      )
+    end
+
+    def create_timeline_entry(details, edition, created_at, created_by = nil)
+      TimelineEntry.create!(
+        entry_type: :whitehall_migration,
+        created_by_id: created_by,
```

Comment:
I'm a bit late to the party here, but this will try to insert the user's Whitehall user ID, rather than their Content Publisher ID (which can be looked up in `user_ids`, i.e. `user_ids[created_by]`).

Reply:
Thanks for this Bruce.  Resolved via PR: https://github.com/alphagov/content-publisher/pull/1738.